lib/peer_manager/modules/peer_data_storage.c

```diff
@@ -56,30 +61,20 @@ static void pds_evt_send(pm_evt_t *p_event)
 	}
 }
 
-/* Function to convert peer IDs to file IDs. */
-static uint16_t peer_id_to_file_id(pm_peer_id_t peer_id)
-{
-	return (uint16_t)(peer_id + PEER_ID_TO_FILE_ID);
-}
-
-/* Function to convert peer data id to type id. */
-static pm_peer_id_t file_id_to_peer_id(uint16_t file_id)
-{
-	return (pm_peer_id_t)(file_id + FILE_ID_TO_PEER_ID);
-}
-
-/* Function to convert peer data IDs to record keys. */
-static uint16_t peer_data_id_to_record_key(pm_peer_data_id_t peer_data_id)
+static uint32_t peer_id_peer_data_id_to_entry_id(pm_peer_id_t peer_id,
+						 pm_peer_data_id_t data_id)
 {
-	return (uint16_t)(peer_data_id + DATA_ID_TO_RECORD_KEY);
+	return (peer_id * PM_PEER_DATA_ID_LAST) + data_id;
 }
 
-/* Function to convert record keys to peer data IDs. */
-static pm_peer_data_id_t record_key_to_peer_data_id(uint16_t record_key)
+static void entry_id_to_peer_id_peer_data_id(uint32_t entry_id, pm_peer_id_t *peer_id,
+					     pm_peer_data_id_t *data_id)
 {
-	return (pm_peer_data_id_t)(record_key + RECORD_KEY_TO_DATA_ID);
+	*data_id = entry_id % PM_PEER_DATA_ID_LAST;
+	*peer_id = entry_id / PM_PEER_DATA_ID_LAST;
 }
 
+#if CODE_DISABLED /* todo: this does not apply anymore. Find an alternative. Used in evt handler. */
```

Comment:
Will the file system be shared with other users? If so, we probably need some explicit namespacing of the entry_id, which would naturally give you the alternative here. If it's not shared, this check is unneeded.

Reply:
That logic is completely obsolete now. In the future we will always be sure that the Peer Manager has its own storage partition. It's one of the refinement I have planned once we iron out all the issues.